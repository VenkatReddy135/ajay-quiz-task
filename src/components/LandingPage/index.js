import React from "react";
import { useNavigate } from "react-router-dom";
import Button from "../../sharedComponent/Button/Button";

const LandingPage = () => {
  const navigate = useNavigate();

  const handleClick = () => {
    navigate(`/questions`);
  };

  return (
    <div className="container" style={{ marginLeft: "40%", marginTop: "22%" }}>
      <Button
        handleClick={handleClick}
        buttonValue="Start Quiz"
        buttonClass="btn btn-warning"
      />
    </div>
  );
};

export default React.memo(LandingPage);
