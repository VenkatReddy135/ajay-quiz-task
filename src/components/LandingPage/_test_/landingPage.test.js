import React from "react";
import { cleanup } from "@testing-library/react";
import ReactDOM from "react-dom";
import renderer from "react-test-renderer";
import LandingPage from "../index";
import Enzyme from "enzyme";
import Adapter from "enzyme-adapter-react-17-updated";
Enzyme.configure({ adapter: new Adapter() });

jest.mock("react-router-dom", () => ({
  useNavigate: jest.fn(),
}));

describe("Landing page component", () => {
  const renderedData = renderer.create(<LandingPage />).toJSON();

  it("renders without crashing", () => {
    const div = document.createElement("div");
    ReactDOM.render(<LandingPage />, div);
  });

  it("renders a snapshot", () => {
    expect(renderedData).toMatchSnapshot();
  });

  afterEach(() => {
    cleanup();
  });
});
