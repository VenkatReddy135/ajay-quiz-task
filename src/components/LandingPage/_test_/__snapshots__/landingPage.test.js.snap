// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Landing page component renders a snapshot 1`] = `
<div
  className="container"
  style={
    Object {
      "marginLeft": "40%",
      "marginTop": "22%",
    }
  }
>
  <button
    className="btn btn-warning"
    onClick={[Function]}
    type="button"
    value=""
  >
    Start Quiz
  </button>
</div>
`;
